Replicator/tests/ReplicatorCollectionSGTest.cc

```diff
@@ -303,21 +303,88 @@ static std::unordered_map<alloc_slice, unsigned> getDocIDs(C4Collection* collect
     return ret;
 }
 
+TEST_CASE_METHOD(ReplicatorCollectionSGTest, "API Push 5000 Changes Collections SG", "[.SyncServerCollection]") {
+    string idPrefix = timePrefix();
+    const string docID = idPrefix + "apipfcc-doc1";
+    
+    string revID;
+    constexpr size_t collectionCount = 1;
+
+    std::array<C4CollectionSpec, collectionCount> collectionSpecs;
+    std::array<C4Collection *, collectionCount> collections;
+    std::array<unordered_map<alloc_slice, unsigned>, collectionCount> docIDs;
+    std::array<C4ReplicationCollection, collectionCount> replCollections;
+
+    collectionSpecs = {
+        Roses
+    };
+    collections = collectionPreamble(collectionSpecs, "sguser", "password");
+    docIDs[0] = getDocIDs(collections[0]);
+    replCollections = {
+        C4ReplicationCollection{collectionSpecs[0], kC4OneShot, kC4Disabled},
+    };
+
+    C4ParamsSetter paramsSetter = [&replCollections,&collectionCount](C4ReplicatorParameters& c4Params) {
+        c4Params.collectionCount = replCollections.size();
+        c4Params.collections = replCollections.data();
+    };
+
+    {
+        TransactionHelper t(db);
+        revID = createNewRev(collections[0], slice(docID), nullslice, kFleeceBody);
+    }
+    
+    replicate(paramsSetter);
+    docIDs[0] = getDocIDs(collections[0]);
+    verifyDocs(collectionSpecs, docIDs);
+
+    C4Log("-------- Mutations --------");
+    {
+        TransactionHelper t(db);
+        for (int i = 2; i <= 5000; ++i)
+            revID = createNewRev(collections[0], slice(docID), slice(revID), kFleeceBody);
+    }
+
+    C4Log("-------- Second Replication --------");
+    replicate(paramsSetter);
+    verifyDocs(collectionSpecs, docIDs);
```

Comment:
verifyDocs can verify the revision generation.The remaining lines are unnecessary.